promql/functions.go

```diff
@@ -996,6 +996,33 @@ func funcHistogramSum(vals []parser.Value, args parser.Expressions, enh *EvalNod
 	return enh.Out
 }
 
+// === histogram_stdvar(Vector parser.ValueTypeVector) Vector ===
+func funcHistogramStdVar(vals []parser.Value, args parser.Expressions, enh *EvalNodeHelper) Vector {
+	inVec := vals[0].(Vector)
+
+	for _, sample := range inVec {
+		// Skip non-histogram samples.
+		if sample.H == nil {
+			continue
+		}
+		mean := sample.H.Sum / sample.H.Count
+		var variance, cVariance float64
+		it := sample.H.AllBucketIterator()
+		for it.Next() {
+			bucket := it.At()
+			mid := (bucket.Upper + bucket.Lower) / 2
+			variance, cVariance = kahanSumInc(bucket.Count*math.Pow(mid-mean, 2), variance, cVariance)
```

Comment:
Not sure how well `math.Pow` is optimized. My gut feeling is that it would be better to do something like `delta := mid-mean` and then write this as
```go
variance, cVariance = kahanSumInc(bucket.Count*delta*delta, variance, cVariance)
```

Or you could not have `mid` at all but calculate the `delta` directly (including the geometric mean idea from above):
```go
delta := mean - math.Sqrt(bucket.Upper * bucket.Lower)
variance, cVariance = kahanSumInc(bucket.Count*delta*delta, variance, cVariance)
```